sycl/source/detail/kernel_impl.cpp

```diff
@@ -93,6 +93,34 @@ bool kernel_impl::isCreatedFromSource() const {
   return MCreatedFromSource;
 }
 
+bool kernel_impl::isBuiltInKernel(const device &Device) const {
+  std::string KernelName = get_info<info::kernel::function_name>();
```

Comment:
Should be moved down below, closer to its use. No need to allocate and query that string if there are no built-in kernels on device.